api/src/main/java/org/eclipse/microprofile/faulttolerance/Asynchronous.java

```diff
@@ -24,21 +24,71 @@
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.util.concurrent.CompletableFuture;
 
 import javax.interceptor.InterceptorBinding;
 
 /**
  * Wrap the execution and invoke it asynchronously.
- * Any methods marked with this annotation must return {@link java.util.concurrent.Future}. 
+ * Any methods marked with this annotation must return one of:
+ * <ul>
+ *   <li>{@link java.util.concurrent.Future}</li>
+ *   <li>{@link java.util.concurrent.CompletionStage}</li>
+ * </ul>
  * Otherwise, {@link org.eclipse.microprofile.faulttolerance.exceptions.FaultToleranceDefinitionException} occurs.
- * Example usage:
  * 
+ * <p>
+ * When a method marked with this annotation is executed, the call returns immediately while the method
+ * is executed in another thread. Returned Future or CompletionStage is not completed until the method 
+ * running asynchronously returns or throws an exception.
+ * </p>
+ * 
+ * <p>After a method marked with this annotation returns normally, the Future or CompletionStage returned
+ * in the calling thread delegate all method calls to the object returned by the method. 
+ * CompletionStage returned in the calling thread is completed when the stage returned by the method is completed
```

Comment:
I think this contradicts the sentence in the previous paragraph: `CompletionStage is not completed until the method running asynchronously returns`.

Reply:
Valid point, I'll change the previous paragraph to something like "The returned Future or CompletionStage is not completed until the result of the method running asynchronously is completed (either normally or exceptionally) or the method throws an exception."

Reply:
Done